spec/ddtrace/contrib/sucker_punch/patcher_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'ddtrace/contrib/support/spec_helper'
+require 'sucker_punch'
+require 'ddtrace'
+require_relative 'dummy_worker'
+
+RSpec.describe 'Sinatra instrumentation' do
+  before do
+    Datadog.configure do |c|
+      c.use :sucker_punch
+    end
+
+    ::SuckerPunch::Queue.clear
+    ::SuckerPunch::RUNNING.make_true
+  end
+
+  after do
+    ::SuckerPunch::Queue.clear
+  end
+
+  around do |example|
+    # Reset before and after each example; don't allow global state to linger.
+    Datadog.registry[:sucker_punch].reset_configuration!
+    example.run
+    Datadog.registry[:sucker_punch].reset_configuration!
+  end
+
+  context 'successful job' do
+    subject(:dummy_work_success) { ::DummyWorker.perform_async }
+
+    it 'should generate two spans, one for pushing to enqueue and one for the job itself' do
+      is_expected.to be true
+      try_wait_until { fetch_spans.length == 2 }
+      expect(spans.length).to eq(2)
+    end
+
+    it 'should instrument successful job' do
+      is_expected.to be true
+      try_wait_until { fetch_spans.length == 2 }
+
+      span = spans.find { |s| s.resource[/PROCESS/] }
+
+      expect('sucker_punch').to eq(span.service)
+      expect('sucker_punch.perform').to eq(span.name)
+      expect('PROCESS DummyWorker').to eq(span.resource)
+      expect('DummyWorker').to eq(span.get_tag('sucker_punch.queue'))
+      expect(Datadog::Ext::Errors::STATUS).not_to eq(span.status)
```

Comment:
A nice helper we have for matching span errors:
```suggestion
      expect(span).to have_error
```

Other span error matchers: https://github.com/DataDog/dd-trace-rb/blob/bac69b5b232317d9b4b3fddeacb2b6766fe9fe21/spec/ddtrace/contrib/excon/instrumentation_spec.rb#L116-L118

Reply:
ah right, nice, updated